text/0201-scope-attributes.md

```diff
@@ -0,0 +1,148 @@
+    data emitted as log records needs to be differentiated so that it can be easily
+    routed and processed differently in the backends. We don't have a good way to handle
+    this today. The type of the data can be recorded as an attribute Logger's Scope.
+- It makes Scope consistent with the other primary data types: Resource, Span, Metric,
+  LogRecord.
+
+See additional [discussion here](https://github.com/open-telemetry/opentelemetry-specification/issues/2450).
+
+## Summary
+
+The following is the summary of proposed changes:
+
+- We will extend OpenTelemetry API to allow specifying Scope attributes when obtaining a
+  Tracer, Meter or LogEmitter. Scope attributes will be optional.
+- We will add `attributes` field to the [InstrumentationScope](https://github.com/open-telemetry/opentelemetry-proto/blob/88faab1197a2a105c7da659951e94bc951d37ab9/opentelemetry/proto/common/v1/common.proto#L83)
+  message of OTLP.
+- We will specify that Telemetry emitted via a Scope-ed Tracer, Meter or LogEmitter will
+  be associated with the Scope's attributes.
+- We will specify that OTLP Exporter will record the attributes in the
+  InstrumentationScope message.
+- We will create a section for Scope attributes' semantic conventions in
+  the specification.
+
+## Internal details
+
+### API Changes
+
+`Get a Tracer` API will be extended to add the following parameter:
+
+```
+- `attributes` (optional): Specifies the instrumentation scope attributes to associate
+  with emitted telemetry.
+```
+
+`Get a Metter` API will be extended to add the following parameter:
+
+```
+- `attributes` (optional): Specifies the instrumentation scope attributes to associate
+  with emitted telemetry.
+```
+
+`Get LogEmitter` SDK call will be altered to the following:
+
+```
+Accepts the instrumentation scope name and optional version and attributes and
+returns a LogEmitter associated with the instrumentation scope.
+```
+
+Since the attributes are optional this is a backwards compatible change.
+
+### OTLP Changes
+
+The InstrumentationScope message in OTLP will be modified to add 2 new fields:
+attributes and dropped_attributes_count:
+
+```protobuf
+message InstrumentationScope {
+  string name = 1;
+  string version = 2;
+  repeated KeyValue attributes = 3;
+  uint32 dropped_attributes_count = 4;
+}
+```
+
+This change is backwards compatible from OTLP's interoperability perspective. Recipients
+of old OTLP versions will not see the Scope attributes and will ignore them, which we
+consider acceptable from interoperability perspective. This is aligned with our general
+stance on what happens when telemetry sources _add_ new data which old recipients
+don't understand: we expect the new data to be safely ignored.
+
+## Attribute Value Precedence
+
+If the same attribute is specified both at the Span/Metric/LogRecord and at the Scope
+then the attribute value at Span/Metric/LogRecord takes precedence.
```

Comment:
The resource is separate from the attributes in the span. But so are the ones in the Scope, as shown above in the protobuf. So I'm not sure why there is a need for precedence at all? If precedence is given to the span/metric/logrecord over the attributes that are separately kept in Scope and potentially Resource then you end up with a situation of allowing the service name to be overridden at the span level.

Reply:
@tsloughter The paragraph just below these lines explains that this is necessary for non-OTLP exporters that "flatten" attributes since non-OTLP formats and consumers are not aware of the hierarchy we have in OTel with Resource vs Span/Metric/LogRecord (vs InstrumentationScope).

Reply:
Ah, of course, thanks. Yea, then needs to define that the Scope attributes take precedence over the Resource attributes, in my opinion --

Reply:
Precedence for Resource attributes and Span attributes must be defined then, right? I went to quickly find it but haven't yet.

Reply:
>Precedence for Resource attributes and Span attributes must be defined then, right? I went to quickly find it but haven't yet.

I can't find it either. The reason is it is not defined in the spec is likely because we did not anticipate that we will find the same attributes on the Resource and on the Span. They each typically have their own semantic conventions that don't overlap.

You are right though that this likely still needs to be defined because it can happen and when flattening a decision must be made one way or another (and Collector indeed does such decisions for some formats, for example [Zipkin](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/1c4dd67f7af084c467ce5378115fa151f7276547/pkg/translator/zipkin/zipkinv2/from_translator.go#L70).

I think that is outside the scope of this OTEP though and can be an independent change in the spec.

Reply:
Added an issue to figure out the Resource precedence: https://github.com/open-telemetry/opentelemetry-specification/issues/2535